### DISCUSSION TITLE
some basic questions - replicating vim/evil and general.el keybindings

### DISCUSSION THREAD

**Discussion by fab6** [meow-edit/meow#160] [Q&A] [answered]
Hi,

I am using vim/spacemacs/evil since many years now and just came across meow. 

It sounds quite interesting though it is not clear to me yet if it is suitable or meant at all to replicate vi-keybindings!? It seems more that one would have other keybindings, but just with a good modal implementation. Is this right?

Maybe there is a setup to replicate evil keybindings with meow!? If so, is there an example setup for this? Would it be possible at all to get vi-keybindings?

Further I understand that it is not possible (or suitable) to use meow and general.el keybindings together...!?

Would be great if you have some comments for my questions.
Thank you in advance!
Fab6

**Comment by DogLooksGood** [comment, marked as answer]
Hi, you can not get a fully compatible setup with evil, since meow use different commands.  You can have some similar behaviors, for example, h/j/k/l to move by character.  But there are still a lot differences.

Meow use keypad state to simulating modified(by ctrl/meta) keys, it doesn't care how those key bindings defined.  There's no integration between meow and general.el.  But if you have a command at `C-c C-c`, not matter how it's bound, you can execute with `SPC c c` via keypad.

Please feel free to ask anything if it's not clear to you, and
Happy New Year!

**Reply by fab6** [reply]
Hi, thank you for the very quick reply and information! I think it is clear to me now... Happy New Year as well!

**Comment by mclearc** [comment]
In case it helps, as I had the same set of questions in moving from vim/evil and general to meow, you might find my settings for [keybindings](https://github.com/mclear-tools/dotemacs/blob/master/setup-config/setup-keybindings.el) and for [meow](https://github.com/mclear-tools/dotemacs/blob/master/setup-config/setup-modal.el) helpful. I basically replaced general with use-package's bind-key, created a leader prefix and a set of namespaced keybinds, and then use meow-leader to trigger the various keymaps. It's much cleaner and simpler than my old evil/general config but get the same results, without clobbering emacs' own keybindings.

**Reply by DogLooksGood** [reply]
If you have custom face for meow, then you can disable dynamic face color.
```
(setq meow-use-dynamic-face-color nil)
```
Make loading faster, and you don't have to care about the load order anymore.

**Reply by mclearc** [reply]
Thanks for the tip!
